homeassistant/components/netatmo/climate.py

```diff
@@ -3,71 +3,128 @@
 from datetime import timedelta
 import voluptuous as vol
 
-from homeassistant.const import TEMP_CELSIUS, ATTR_TEMPERATURE
+from homeassistant.const import (
+    STATE_OFF, TEMP_CELSIUS, ATTR_TEMPERATURE, STATE_UNKNOWN, CONF_NAME)
 from homeassistant.components.climate import ClimateDevice, PLATFORM_SCHEMA
 from homeassistant.components.climate.const import (
-    STATE_HEAT, STATE_IDLE,
-    SUPPORT_TARGET_TEMPERATURE, SUPPORT_OPERATION_MODE, SUPPORT_AWAY_MODE)
+    STATE_HEAT, STATE_IDLE, SUPPORT_ON_OFF, SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_OPERATION_MODE, SUPPORT_AWAY_MODE, STATE_MANUAL, STATE_AUTO,
+    STATE_ECO, STATE_COOL)
 from homeassistant.util import Throttle
 import homeassistant.helpers.config_validation as cv
 
+
 DEPENDENCIES = ['netatmo']
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_RELAY = 'relay'
-CONF_THERMOSTAT = 'thermostat'
+CONF_HOMES = 'homes'
+CONF_ROOMS = 'rooms'
 
-DEFAULT_AWAY_TEMPERATURE = 14
-# # The default offset is 2 hours (when you use the thermostat itself)
-DEFAULT_TIME_OFFSET = 7200
-# # Return cached results if last scan was less then this time ago
-# # NetAtmo Data is uploaded to server every hour
-MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=300)
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_RELAY): cv.string,
-    vol.Optional(CONF_THERMOSTAT, default=[]):
-        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_HOMES): vol.All(cv.ensure_list, [
+        {
```

Comment:
For readability, it would be nice if you could move the "Home" object to it's own Schema. See this example for details: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/zoneminder/__init__.py#L25

Reply:
Thanks for your advice. However, I failed the follow the example you gave. In my case I have to use `PLATFORM_SCHEMA.extend` but after that the configuration cannot be retrieved as before. (Reporting `AttributeError: 'str' object has no attribute 'get'`, when trying to retrieve the nested configuration of each Home). Could you please advise how to do this correctly?

Reply:
HOME_CONFIG_SCHEMA = vol.Schema({
    vol.Required(CONF_NAME): cv.string,
    vol.Optional(CONF_ROOMS, default=[]): vol.All(cv.ensure_list, [cv.string])
})

PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
    vol.Optional(CONF_HOMES): vol.All(cv.ensure_list, [HOME_CONFIG_SCHEMA])
})

Reply:
Thanks! Now it has been updated as you suggested: 0dbe938.